File: android/src/main/java/org/ligi/gobandroid_hd/ui/UndoWithVariationDialog.java
package org.ligi.gobandroid_hd.ui;

import android.content.Context;
import android.content.DialogInterface;
import android.widget.CheckBox;

import org.ligi.gobandroid_hd.App;
import org.ligi.gobandroid_hd.R;

public class UndoWithVariationDialog extends GobandroidDialog {

    public UndoWithVariationDialog(Context context) {
        super(context);
        setTitle(R.string.keep_variant_);
        setIconResource(R.drawable.help);
        setContentView(R.layout.dialog_keep_variant);

        final CheckBox prevent_cb = (CheckBox) findViewById(R.id.keep_variant_session_cb);

        class OnYesClick implements OnClickListener {

            @Override
            public void onClick(DialogInterface dialog, int which) {
                App.getGame().undo(true);
                if (prevent_cb.isChecked()) {
                    App.getInteractionScope().ask_variant_session = false;
                }

                dialog.dismiss();
            }

        }

        class OnNoClick implements OnClickListener {

            @Override
            public void onClick(DialogInterface dialog, int which) {
                App.getGame().undo(false);
                if (prevent_cb.isChecked()) {
                    App.getInteractionScope().ask_variant_session = false;
                }

                dialog.dismiss();
            }

        }

        setPositiveButton(R.string.yes, new OnYesClick());
        setNegativeButton(R.string.no, new OnNoClick());
    }

}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/UndoWithVariationDialog.kt
package org.ligi.gobandroid_hd.ui

import android.app.Activity
import android.support.design.widget.Snackbar

import org.ligi.gobandroid_hd.InteractionScope
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.logic.GoGame

class UndoWithVariationDialog private constructor(activity: Activity) : GobandroidDialog(activity) {

    companion object {

        fun userInvokedUndo(activity: Activity, interactionScope: InteractionScope, game: GoGame) {
            if (interactionScope.ask_variant_session) {
                UndoWithVariationDialog(activity).show()
            } else {
                game.undo(GoPrefs.isKeepVariantWanted)

                val snackbar = Snackbar.make(activity.window.decorView.findViewById(R.id.content_frame), R.string.snackbar_keep_variant_message_enable, Snackbar.LENGTH_LONG)
                snackbar.setAction(R.string.snackbar_keep_variant_action_enable) { interactionScope.ask_variant_session = true }
                snackbar.show()

            }
        }
    }

    init {
        setTitle(R.string.keep_variant)
        setIconResource(R.drawable.ic_action_help_outline)
        setContentView(R.layout.dialog_keep_variant)

        setPositiveButton(R.string.yes, { dialog ->
            gameProvider.get().undo(true)
            dialog.dismiss()
        })

        setNegativeButton(R.string.no, { dialog ->
            gameProvider.get().undo(false)
            dialog.dismiss()

            val snackbar = Snackbar.make(activity.window.decorView.findViewById(R.id.content_frame), R.string.snackbar_keep_variant_message_disable, Snackbar.LENGTH_LONG)
            snackbar.setAction(R.string.snackbar_keep_variant_action_disable) { interactionScope.ask_variant_session = false }
            snackbar.show()
        })
    }


}
